ec/ecresource/deploymentresource/elasticsearch/v2/topology_plan_modifier.go

```diff
@@ -23,59 +23,77 @@ import (
 	"github.com/hashicorp/terraform-plugin-framework/attr"
 	"github.com/hashicorp/terraform-plugin-framework/diag"
 	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
 	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
 )
 
 // Use current state for a topology's attribute if the topology's state is not nil and the template attribute has not changed
-func UseTopologyStateForUnknown(topologyAttributeName string) tfsdk.AttributePlanModifier {
+func UseTopologyStateForUnknown(topologyAttributeName string) useTopologyState {
```

Comment:
Similarly, might be worth adding some type assertions for this struct at the top